x11/tk/Portfile

```
@@ -25,82 +25,130 @@ checksums           md5 6d64b6eb021062f378017d403fedcbe6 \
+    set incdir          ${prefix}/include/${subport}
+    set libdir          ${prefix}/lib/${subport}
+    set mandir          ${prefix}/share/${subport}/man
+
+    configure.args      --bindir=${bindir} \
+                        --includedir=${incdir} \
+                        --libdir=${libdir} \
+                        --mandir=${mandir} \
+                        --with-tcl=${prefix}/lib
+    configure.env       {TK_LIBRARY=$(libdir)/tk$(VERSION)}
+    platform darwin {
         configure.args-append \
-                    --enable-aqua
-        post-destroot {
-            move \
-                ${destroot}${prefix}/include/X11 \
-                ${destroot}${prefix}/include/X11_tk
-            fs-traverse fl ${destroot}${prefix}/include {
-                if {[file extension ${fl}] eq ".h"} {
-                    reinplace -q "s|include <X11/|include <X11_tk/|g" ${fl}
-                    reinplace -q "s|include \"X11/|include \"X11_tk/|g" ${fl}
-                }
-            }
+                        tcl_cv_type_64bit="long long"
+    }
+    # see https://trac.macports.org/ticket/58447
+    configure.cppflags-replace -I${prefix}/include -isystem${prefix}/include
+    configure.ldflags   -L${worksrcpath} -L${prefix}/lib
+
+    configure.args.arm64-append     --enable-64bit
+    configure.args.x86_64-append    --enable-64bit
+    configure.args.ppc64-append     --enable-64bit
+
+    # see https://trac.macports.org/ticket/17189
+    destroot.target-append \
+                        install-private-headers
+    post-destroot {
+        ln -s wish8.6 ${destroot}${bindir}/wish
+        ln -s libtk8.6.dylib ${destroot}${libdir}/libtk.dylib
+    
+        foreach fl {tk3d.h tkFont.h} {
+            xinstall -m 0644 ${workpath}/tk${version}/generic/${fl} ${destroot}${incdir}
         }
     }
 
-    variant x11 conflicts quartz {}
+    test.run            yes
 
-    if {![variant_isset x11]} {
-        default_variants +quartz
-    }
-} else {
-    # Even though X11 is the only option, make a variant so that other ports
-    # can check for it being set with require_active_variants.
-    variant x11 {}
-    default_variants +x11
+    livecheck.type      none
 }
 
-platform darwin {
+subport tk-quartz {
+    platforms       {macosx >= 10}
+    supported_archs arm64 i386 x86_64
     configure.args-append \
-                    tcl_cv_type_64bit="long long"
-}
-
-post-destroot {
-    foreach fl {tk3d.h tkFont.h} {
-        xinstall -m 0644 ${workpath}/tk${version}/generic/${fl} ${destroot}${prefix}/include/
+                    --enable-aqua
+    post-destroot {
+        move \
+            ${destroot}${incdir}/X11 \
+            ${destroot}${incdir}/X11_tk
+        fs-traverse fl [list ${destroot}${incdir}] {
+            if {[file extension ${fl}] eq ".h"} {
+                reinplace -q "s|include <X11/|include <X11_tk/|g" ${fl}
+                reinplace -q "s|include \"X11/|include \"X11_tk/|g" ${fl}
```

Since you're replacing in a bunch of files, it would be faster if you did both replacements at once.
```suggestion
                reinplace -E -q {s|(include [<"]X11)/|\1_tk/|g} ${fl}
```

---

Fair point, but probably outside the scope of this PR since it's preexisting code from the quartz variant.